### PR TITLE
NCC-77 Add message about future NVS integration with plugin

### PR DESCRIPTION
**Pull request by ndonewar**
https://issues.sonatype.org/browse/NCC-77

This PR is for adding a message to project pages about a possible NVS integration in the Jenkins plugin. The message does not show up if IQ is configured. The message is dismissible by users (and stays hidden via cookie). It can also be hidden globally by administrators via the Manage Jenkins configuration. The message and configuration will be removed in the future (once decisions are made to pursue or abandon NVS integration).

Screenshot of NVS message:

NVS Message

Screenshot of configuration:

Global configuration

### CODE REVIEW

File: src/main/java/org/sonatype/nexus/ci/nvs/NvsMessageAction.groovy [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.nvs
+
+import javax.annotation.CheckForNull
+
+import hudson.model.Action
+
```
Comment (derickfernando) [inline src/main/java/org/sonatype/nexus/ci/nvs/NvsMessageAction.groovy]: Would be nice if we add a comment here about why this implementation returns null.

File: src/main/resources/org/sonatype/nexus/ci/nvs/NvsMessageAction/jobMain.groovy [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.nvs.NvsMessageAction
+
+def t = namespace(lib.JenkinsTagLib)
+
+style('''
+  #nexus-plugin-nvs-message {
+    display: none;
```
Comment (derickfernando) [inline src/main/resources/org/sonatype/nexus/ci/nvs/NvsMessageAction/jobMain.groovy]: +1

File: src/main/java/org/sonatype/nexus/ci/nvs/NvsMessageAction.groovy (hunk 1 above)
Comment (ndonewar) [reply src/main/java/org/sonatype/nexus/ci/nvs/NvsMessageAction.groovy]: Good idea - I updated and copied the comment from `PolicyEvaluationProjectAction`

**Comment by whyjustin** [comment]
- There are some alignment issues, the icon in the job page and perhaps the checkbox too?
- Have you chatted with @jeviolle @tneer about where they would expect this kind of notification would appear? While it seems pretty invasive to appear on every job page, you have provided the ability to hide per user and globally. Regardless the CDI team may have suggestions of better places or better practices in Jenkins that aren't as invasive.

**Comment by ndonewar** [comment]
@whyjustin Thanks for the comments

1. I'll reach out to Rick and Troy for their thoughts
2. The alignments bothered me, but the issues come from Jenkins.
a. The icons for Workspace and Recent Changes have extra padding on their left side:Screen Shot 2019-04-16 at 1 19 58 PM
b. And the Sonatype logo gets extra padding on its right side from Jenkins: Screen Shot 2019-04-16 at 1 20 34 PM
c. I thought about using CSS to override the defaults, but thought that could cause misalignment with other plugins. The logo for the IQ integration doesn't change the default spacing, so I went with that approach. I'm open to revisiting it though, if you still think we should. Screen Shot 2019-04-16 at 1 27 41 PM
d. As for the checkbox alignment, we're using the standard form controls. Generally all checkboxes are misaligned with the rest of the controls. Here's an example from Jenkins itself: Screen Shot 2019-04-16 at 1 21 23 PM

**Comment by whyjustin** [comment]
> 2\. The logo for the IQ integration doesn't change the default spacing, so I went with that approach. I'm open to revisiting it though, if you still think we should.

Heh, be careful where I stick my nose. There's probably examples of other plugins to understand precedent. https://github.com/search?l=&q=org%3Ajenkinsci+filename%3AjobMain.groovy&type=Code

**Comment by ndonewar** [comment]
@whyjustin That's interesting - most of the ones using the summary icon do add the space. I'm going to update ours. Thanks!

**Comment by ndonewar** [comment]
Closing to open in upstream

**Pull request closed without merging**